hack/test/unit

```diff
@@ -12,22 +12,42 @@
 #
 set -eux -o pipefail
 
-BUILDFLAGS=(-tags 'netgo seccomp libdm_no_deferred_remove')
-TESTFLAGS+=" -test.timeout=${TIMEOUT:-5m}"
+# run_test_unit runs unit tests for packages selected.
+# arguments:
+# $1 - prefix to use in generated file names. For example, '-libnetwork'.
+# $2 - packages to test
+run_test_unit() {
+	mkdir -p bundles
+	gotestsum --format=standard-quiet --jsonfile=bundles/go-test-report${1}.json --junitfile=bundles/junit-report${1}.xml -- \
+		-tags 'netgo seccomp libdm_no_deferred_remove' \
+		-cover \
+		-coverprofile=bundles/profile${1}.out \
+		-covermode=atomic \
+		${TESTFLAGS:-} \
+		-test.timeout=${TIMEOUT:-5m} \
+		${2}
+}
+
 TESTDIRS="${TESTDIRS:-./...}"
 exclude_paths='/vendor/|/integration'
 pkg_list=$(go list $TESTDIRS | grep -vE "($exclude_paths)")
 
-echo "${pkg_list}" | grep --fixed-strings "libnetwork/drivers/bridge" \
+# NOTE: libnetwork tests require sequential execution.
+# See https://github.com/moby/moby/issues/42458#issuecomment-873216754 for more info.
+# All packages in `/libnetwork` namespace are extracted into a separate package set to account for that below.
+
+base_pkg_list=$(echo "${pkg_list}" | grep --fixed-strings -v "/libnetwork" || :)
+libnetwork_pkg_list=$(echo "${pkg_list}" | grep --fixed-strings "/libnetwork" || :)
+
+echo "${libnetwork_pkg_list}" | grep --fixed-strings "libnetwork/drivers/bridge" \
 	&& if ! type docker-proxy; then
 		hack/make.sh binary-proxy install-proxy
 	fi
 
-mkdir -p bundles
-gotestsum --format=standard-quiet --jsonfile=bundles/go-test-report.json --junitfile=bundles/junit-report.xml -- \
-	"${BUILDFLAGS[@]}" \
-	-cover \
-	-coverprofile=bundles/profile.out \
-	-covermode=atomic \
-	${TESTFLAGS} \
-	${pkg_list}
+if [ "${base_pkg_list}" ]; then
+	run_test_unit '' "${base_pkg_list}"
+fi
+if [ "${libnetwork_pkg_list}" ]; then
+	# Run only one test binary in parallel to ensure sequential execution.
+	TESTFLAGS="-p=1 ${TESTFLAGS:-}" run_test_unit '-libnetwork' "${libnetwork_pkg_list}"
```

Comment:
Hm.. I'm a bit on the fence on the second commit abstracting away things into `run_test_unit` - as seen here, it still needs various parameters to be passed (test flags, prefix, package-list), but now it requires me to look back-and-forth between this line and the function above to understand what each of those do.

I think it's fine to keep the code inline here. It's not "pretty", but reduces the cognitive overhead of having to look up what `run_test_unit` does, and what parameters it takes.